nomos-blend/scheduling/src/message_scheduler/mod.rs

```diff
@@ -260,12 +219,10 @@ where
                 if Box::pin(setup_new_session(
```

Comment:
Nice! Making this non `async` solves a big problem, so we can just get rid of polling from inside here, which would otherwise require an additional synchronization primitive to avoid polling a new session before the previous setup was completed, which I took notes and wanted to resolve on Monday.